x-pack/plugins/observability/public/pages/alerts/components/observability_actions.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonIcon,
+  EuiFlexItem,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import React, { useMemo, useState, useCallback } from 'react';
+
+import { CaseAttachments } from '@kbn/cases-plugin/public';
+import { CommentType } from '@kbn/cases-plugin/common';
+import type { ActionProps } from '@kbn/timelines-plugin/common';
+import { useKibana } from '../../../utils/kibana_react';
+import { observabilityFeatureId } from '../../../../common';
+import { useGetUserCasesPermissions } from '../../../hooks/use_get_user_cases_permissions';
+import { parseAlert } from './parse_alert';
+import { translations, paths } from '../../../config';
+import {
+  ADD_TO_EXISTING_CASE,
+  ADD_TO_NEW_CASE,
+} from '../containers/alerts_table_t_grid/translations';
+import { ObservabilityAppServices } from '../../../application/types';
+import { RULE_DETAILS_PAGE_ID } from '../../rule_details/types';
+import type { TopAlert } from '../containers/alerts_page/alerts_page';
+import { ObservabilityRuleTypeRegistry } from '../../..';
+
+export type ObservabilityActionsProps = Pick<
+  ActionProps,
+  'data' | 'eventId' | 'ecsData' | 'setEventsDeleted'
+> & {
+  setFlyoutAlert: React.Dispatch<React.SetStateAction<TopAlert | undefined>>;
+  observabilityRuleTypeRegistry: ObservabilityRuleTypeRegistry;
+  id?: string;
+};
+
+export function ObservabilityActions({
+  data,
+  eventId,
+  ecsData,
+  id: pageId,
```

Comment:
nit: I find it easier to follow when the `id` is renamed `pageId`. Do we want to do something similar in the `AlertsFlyout` component as well?